src/ray/object_manager/object_manager.cc

```diff
@@ -409,6 +410,59 @@ void ObjectManager::PushLocalObject(const ObjectID &object_id, const NodeID &nod
                      std::move(release_chunk_callback));
 }
 
+void ObjectManager::PushSpilledObject(const ObjectID &object_id, const NodeID &node_id,
+                                      const std::string &spilled_url) {
+  // SpilledObject::CreateSpilledObject does synchronous IO; schedule it off
+  // main thread.
+  rpc_service_.post(
+      [this, object_id, node_id, spilled_url, chunk_size = config_.object_chunk_size]() {
+        auto optional_spilled_object =
+            SpilledObject::CreateSpilledObject(spilled_url, chunk_size);
+        if (!optional_spilled_object.has_value()) {
+          RAY_LOG(WARNING) << "Failed to load splled object " << object_id
```

Comment:
```suggestion
          RAY_LOG(ERROR) << "Failed to load spilled object " << object_id
```

I'm wondering if the program will hang if we hit this case. Ideally, we'd return an error message to the caller.

We could test it by manually deleting the spill files during execution.

Reply:
let me see if I can have some tests for this case

Reply:
confirmed it hangs...